dotCMS/src/main/java/com/dotcms/tika/TikaUtils.java

```diff
@@ -138,6 +151,124 @@ public boolean generateMetaData(Contentlet contentlet)
+            throws DotSecurityException, DotDataException {
+
+        return this.generateMetaData(contentlet, binaryField, metadataFields, false);
+    }
+
+    @CloseDBIfOpened
+    private Map<String, Object>  generateMetaData(final Contentlet contentlet, final File binaryField,
+                                                 final Set<String> metadataFields, final boolean force)
+            throws DotSecurityException, DotDataException {
+
+        //See if we have content metadata file
+        Map<String, Object> metaDataMap = Collections.emptyMap();
+        final File contentMetaFile      = APILocator.getFileAssetAPI().
+                getContentMetadataFile(contentlet.getInode());
+
+        /*
+        If we want to force the parse of the file and the generation of the metadata file
+        we need to delete the existing one first.
+         */
+        if (force && contentMetaFile.exists()) {
+            try {
+                contentMetaFile.delete();
+            } catch (Exception e) {
+                Logger.error(this.getClass(),
+                        String.format("Unable to delete existing metadata file [%s] [%s]",
+                                contentMetaFile.getAbsolutePath(), e.getMessage()), e);
+            }
+        }
+
+        //If the metadata file does not exist we need to parse and get the metadata for the file
+        if (!contentMetaFile.exists()) {
+
+            if (binaryField != null) {
+
+                //Parse the metadata from this file
+                metaDataMap =
+                        this.getForcedMetaDataMap(binaryField, metadataFields,
+                                Config.getIntProperty("binaryfield.metadata.content.maxlength", 1024));
+
+                this.writeCompressJsonMetadataFile (contentMetaFile,
+                        UtilMethods.isSet(metaDataMap)?metaDataMap:Collections.emptyMap());
+            }
+        } else {
+
+            metaDataMap = this.readCompressedJsonMetadataFile (contentMetaFile);
+        }
+
+        return metaDataMap;
+    }
+
+
+
+    private Map<String, Object> readCompressedJsonMetadataFile(final File contentMetaFile) {
+
+        Map<String, Object> objectMap = Collections.emptyMap();
+        // compressor config
+        final String compressor = Config
+                .getStringProperty("CONTENT_METADATA_COMPRESSOR", "none");
+
+        try (InputStream inputStream = FileUtil.createInputStream(contentMetaFile.toPath(), compressor)) {
+
+            objectMap   = objectMapper.readValue(inputStream, Map.class);
+            Logger.info(this, "Metadata read from: " + contentMetaFile);
+        } catch (IOException e) {
+
+            Logger.error(this, e.getMessage(), e);
+        }
+
+        return objectMap;
+    }
+
+    private void writeCompressJsonMetadataFile(final File contentMetaFile, final Map<?, Object> objectMap) {
+
+        OutputStream out = null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this OutputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4960665)